File: app/src/main/java/lt/markmerkk/entities/IDataListener.java
package lt.markmerkk.entities;

import javafx.collections.ObservableList;

/**
 * A callback listener when some event occurs on
 * storage
 */
public interface IDataListener<T> {
  /**
   * Called when new data is available
   * @param data
   */
  void onDataChange(ObservableList<T> data);

}

File: components/src/main/java/lt/markmerkk/IDataListener.kt
package lt.markmerkk

import javafx.collections.ObservableList

/**
 * A callback listener when some event occurs on
 * storage
 */
interface IDataListener<T> {
    /**
     * Called when new data is available
     * @param data
     */
    fun onDataChange(data: ObservableList<T>)

}
